lib/brain.js

```diff
@@ -900,18 +906,22 @@ Brain.prototype.smsCompliance = function smsCompliance (nextScreen) {
   flow.on('success', () => {
     const phone = flow.phone
     this.flow = null
-
     this.tx = Tx.update(this.tx, {phone})
-    this.startScreen(true)
+
+    if(returnState) {
```

Comment:
Lots of flows are not handled correctly. For instance, if you enter in the wrong security code 3 times, you will be sent back to the start screen instead of being given a chance to send your coins. These flows have to modified from the original code which was only designed to work with a threshold of 0.